code/modules/mob/mob_helpers.dm

```diff
@@ -73,10 +73,9 @@ mob/proc/remove_internal_organ()
 		return .
 	else if(has_reagent_in_blood(DETCOFFEE))
 		return NOIRMATRIX
-	else if(get_item_by_slot(slot_glasses) && istype(get_item_by_slot(slot_glasses), /obj/item/clothing/glasses/scanner))
-		var/obj/item/clothing/glasses/scanner/S = get_item_by_slot(slot_glasses)
-		if(S.on && S.color_matrix)
-			return S.color_matrix
+	var/obj/item/clothing/glasses/scanner/S = is_wearing_item(slot_glasses)
```

Comment:
you need to specify the type you're looking for to `is_wearing_item` or this will runtime for `glasses` that are not `scanner`s